Mathlib/Data/Polynomial/Derivation.lean

```diff
@@ -92,4 +92,59 @@ def mkDerivationEquiv : A ≃ₗ[R] Derivation R R[X] A :=
 @[simp] lemma mkDerivationEquiv_symm_apply (D : Derivation R R[X] A) :
     (mkDerivationEquiv R).symm D = D X := rfl
 
+end CommSemiring
 end Polynomial
+
+namespace Derivation
+
+variable {R A M : Type*} [CommSemiring R] [CommSemiring A] [Algebra R A] [AddCommMonoid M]
+  [Module A M] [Module R M] [IsScalarTower R A M] (d : Derivation R A M) (a : A)
+
+open Polynomial Module
+
+/--
+For a derivation `d : A → M` and an element `a : A`, `d.compAEval a` is the
+derivation of `R[X]` which takes a polynomial `f` to `d(aeval a f)`.
+
+This derivation takes values in `Module.AEval R M a`, which is `M`, regarded as an
+`R[X]`-module, with the action of a polynomial `f` defined by `f • m = (aeval a f) • m`.
+-/
+/-
+Note: It is currently not easy to define `compAEval` using `Derivation.compAlgebraMap`.
+This is partly because `A` is not an `R[X]` algebra and partly because `Derivation.compAlgebraMap`
+is only implemented in the case that `R` is a `CommRing`, not a `CommSemiring`.
```

Comment:
#8151

Reply:
Thank you very much for pointing this out; I've updated my comment.
Can I check whether you agree with the comment now, or should the definition be changed?

Reply:
Updating the comment is enough, thanks!